### PR TITLE
advancedtls: Rename RevocationConfig

### PR DESCRIPTION
**Pull request by gtcooke94**
This PR renames `RevocationConfig` to `RevocationOptions` to match the naming of all of our other `Options` structs. A type alias remains as to not break existing code.

RELEASE NOTES: none

### CODE REVIEW

File: security/advancedtls/advancedtls.go [hunk 1]
```diff
@@ -415,12 +415,12 @@ func (o *ServerOptions) config() (*tls.Config, error) {
 // advancedTLSCreds is the credentials required for authenticating a connection
 // using TLS.
 type advancedTLSCreds struct {
-	config           *tls.Config
-	verifyFunc       CustomVerificationFunc
-	getRootCAs       func(params *GetRootCAsParams) (*GetRootCAsResults, error)
-	isClient         bool
-	verificationType VerificationType
```
Comment (erm-g) [inline security/advancedtls/advancedtls.go:422]: Any reason why you changed the order here (and across the test)?

File: security/advancedtls/crl.go [hunk 2]
```diff
@@ -55,8 +55,9 @@ type Cache interface {
 	Get(key any) (value any, ok bool)
 }
 
-// RevocationConfig contains options for CRL lookup.
-type RevocationConfig struct {
+// RevocationOptions allows a user to configure revocation using certificate
```
Comment (erm-g) [inline security/advancedtls/crl.go:58]: I'd word it like 'RevocationOptions allows users to configure certification revocation behavior' - WDYT?

File: security/advancedtls/advancedtls.go (hunk 1 above)
Comment (gtcooke94) [reply security/advancedtls/advancedtls.go:422]: Copy/paste changes, I can change it to retain the ordering if you think it's worth it

File: security/advancedtls/crl.go (hunk 2 above)
Comment (gtcooke94) [reply security/advancedtls/crl.go:58]: I like that more, changed

File: security/advancedtls/advancedtls.go (hunk 1 above)
Comment (erm-g) [reply security/advancedtls/advancedtls.go:468]: Oh ok I thought you wanted to sort it alphabetically

**Review by erm-g — APPROVED** [review]

**Review by arvindbr8 — APPROVED** [review]

**Pull request merged**